packages/sdk/src/oracle/services/statemachines/setActiveRequestByTransaction.ts

```diff
@@ -0,0 +1,84 @@
+import assert from "assert";
+import { Update } from "../update";
+import Store from "../../store";
+import { Log, LogDescription } from "../../types/ethers";
+import { Handlers as GenericHandlers } from "../../types/statemachine";
+import { optimisticOracle } from "../../../clients";
+
+// required exports for state machine
+export type Params = { chainId: number; transactionHash: string; eventIndex?: number };
+export type Memory = { error?: Error };
+export function initMemory(): Memory {
+  return {};
+}
+export function Handlers(store: Store): GenericHandlers<Params, Memory> {
+  const update = new Update(store);
+  return {
+    async start(params: Params, memory: Memory) {
+      const { chainId, transactionHash, eventIndex = 0 } = params;
+
+      // have to do all of this to fetch the identifier, ancData, requester and timestamp from the request
+      const provider = store.read().provider(chainId);
+      const receipt = await provider.getTransactionReceipt(transactionHash);
+      const oracleLogs = receipt.logs.reduce((result: LogDescription[], log: Log) => {
+        try {
+          result.push(optimisticOracle.contractInterface.parseLog(log));
+        } catch (err) {
+          // we can ignore this error
```

Comment:
Prefer not using try-catch for normal/expected branching.

An example of how this could cause issues. If we were to have slightly different versions of the OptimisticOracle (or the SkinnyOptimisticOracle) and we ended up using an interface where the event definitions in our abi here didn't perfectly match the contract on-chain, this would silently fail. We've seen this issue in the past when we've made minor updates to contracts on-chain.

Using try-catch this way is sometimes the _only_ option, but very often there's a way to avoid using errors for branching.

In this case, the `Log` type contains an address field that tells us what contract emitted the event, so we could pre-filter the events we parse to only those emitted by the OptimisticOracle contract that we care about. If any events left after the filtering fail to parse, we know that our abi/address is wrong or something else went wrong in the parsing process.

Reply:
Great point. Will update this

Reply:
updated this, first filters for event originating from oracle address, then maps them with the log decoder, its a lot cleaner this way and we dont need to ignore errors